drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/AbstractScenarioSimulationDocksHandler.java

```diff
@@ -0,0 +1,164 @@
+                                        perspectiveIdentifier);
+        result.add(settingsDock.withSize(450).withLabel(ScenarioSimulationEditorConstants.INSTANCE.settings()));
+        toolsDock = new UberfireDock(UberfireDockPosition.EAST,
+                                     "INFO_CIRCLE",
+                                     new DefaultPlaceRequest(TestToolsPresenter.IDENTIFIER),
+                                     perspectiveIdentifier);
+        result.add(toolsDock.withSize(450).withLabel(ScenarioSimulationEditorConstants.INSTANCE.testTools()));
+        cheatSheetDock = new UberfireDock(UberfireDockPosition.EAST,
+                                          "FILE_TEXT",
+                                          new DefaultPlaceRequest(CheatSheetPresenter.IDENTIFIER),
+                                          perspectiveIdentifier);
+        result.add(cheatSheetDock.withSize(450).withLabel(ScenarioSimulationEditorConstants.INSTANCE.scenarioCheatSheet()));
+        return result;
+    }
+
+    public void addDocks() {
+        refreshDocks(true, false);
+    }
+
+    public void removeDocks() {
+        refreshDocks(true, true);
+    }
+
+    public void expandToolsDock() {
+        authoringWorkbenchDocks.expandAuthoringDock(toolsDock);
+    }
+
+    public abstract void expandTestResultsDock();
+
+    public void resetDocks() {
+        getSettingsPresenter().ifPresent(SubDockView.Presenter::reset);
+        getCheatSheetPresenter().ifPresent(SubDockView.Presenter::reset);
+        getTestToolsPresenter().ifPresent(SubDockView.Presenter::reset);
+    }
+
+    public void setScesimEditorId(String scesimEditorId) {
+        currentScesimEditorId = scesimEditorId;
+        settingsDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+        toolsDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+        cheatSheetDock.getPlaceRequest().addParameter(SCESIMEDITOR_ID, scesimEditorId);
+    }
+
+    public Optional<CheatSheetView.Presenter> getCheatSheetPresenter() {
+        final Optional<CheatSheetView> cheatSheetView = getCheatSheetView(getCurrentRightDockPlaceRequest(CheatSheetPresenter.IDENTIFIER));
+        return cheatSheetView.map(CheatSheetView::getPresenter);
+    }
+
+    public Optional<TestToolsView.Presenter> getTestToolsPresenter() {
+        final Optional<TestToolsView> testToolsView = getTestToolsView(getCurrentRightDockPlaceRequest(TestToolsPresenter.IDENTIFIER));
+        return testToolsView.map(TestToolsView::getPresenter);
+    }
+
+    public Optional<SettingsView.Presenter> getSettingsPresenter() {
+        final Optional<SettingsView> settingsView = getSettingsView(getCurrentRightDockPlaceRequest(SettingsPresenter.IDENTIFIER));
+        return settingsView.map(SettingsView::getPresenter);
+    }
+
+    /**
+     * Returns a <code>PlaceRequest</code> for the <b>status</b> of the right dock with the given <b>identifier</b>
```

Comment:
Would be nice to keep it according to javadoc reccomendations.
Like, describtion of function and next the @return tag and description of return

Reply:
@dupliaka done!